### PR TITLE
Group Key Management cluster XML naming and types updated according t…

### LINKED ISSUE #13176: Group key management cluster XML looks nothing like the spec
#### Problem
The XML bits added in https://github.com/project-chip/connectedhomeip/pull/12479 look nothing like the spec.  The naming is off.  The integer types used are wrong in a bunch of places (e.g. hardcoding sized ints for things like fabric indices, group ids, etc, some of them with the wrong sizes).  Times are not using the `epoch-us` type, which is what the spec has.  Nullable fields are not marked nullable.  Attributes that are supposed to be marked writable are not marked writable.

 #### Proposed Solution
Go through and make the XML match the spec.

### PR DESCRIPTION
**Pull request by rcasallas-silabs**
#### Problem
* Fixes [#13176](https://github.com/project-chip/connectedhomeip/issues/13176) 

#### Change overview
* Group Key Management cluster XML (src/app/zap-templates/zcl/data-model/chip/group-key-mgmt-cluster.xml) updated.
* Code auto-generated.

#### Testing
* Project built successfully.

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
fast track: xml data type change only, created and approved by domain owner.

File: src/app/zap-templates/zcl/data-model/chip/group-key-mgmt-cluster.xml [hunk 1]
```diff
@@ -17,31 +17,31 @@ limitations under the License.
...
+  <struct name="GroupInfoMapStruct">
     <cluster code="0x003F"/>
-    <item fieldId="0" name="fabricIndex" type="INT16U"/>
-    <item fieldId="1" name="groupId" type="INT16U"/>
-    <item fieldId="2" name="endpoints" type="INT16U" array="true"/>
-    <item fieldId="3" name="groupName" type="CHAR_STRING" length="16" />
+    <item fieldId="0" name="FabricIndex" type="fabric_idx"/>
+    <item fieldId="1" name="GroupId" type="group_id"/>
+    <item fieldId="2" name="Endpoints" type="endpoint_no" array="true"/>
+    <item fieldId="3" name="GroupName" type="CHAR_STRING" length="16" />
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/group-key-mgmt-cluster.xml:32]: And this field should be marked optional.

File: src/app/zap-templates/zcl/data-model/chip/group-key-mgmt-cluster.xml [hunk 2]
```diff
@@ -17,31 +17,31 @@ limitations under the License.
...
-    <item fieldId="6" name="epochKey2" type="OCTET_STRING" length="16"/>
-    <item fieldId="7" name="epochStartTime2" type="INT64U"/>
+    <item fieldId="0" name="GroupKeySetID" type="INT16U"/>
+    <item fieldId="1" name="GroupKeySecurityPolicy" type="GroupKeySecurityPolicy"/>
+    <item fieldId="2" name="EpochKey0" type="OCTET_STRING" length="16"/>
+    <item fieldId="3" name="EpochStartTime0" type="epoch_us"/>
+    <item fieldId="4" name="EpochKey1" type="OCTET_STRING" length="16"/>
+    <item fieldId="5" name="EpochStartTime1" type="epoch_us"/>
+    <item fieldId="6" name="EpochKey2" type="OCTET_STRING" length="16"/>
+    <item fieldId="7" name="EpochStartTime2" type="epoch_us"/>
```
[the remark this answers is not in the document]
Comment (rcasallas-silabs) [reply src/app/zap-templates/zcl/data-model/chip/group-key-mgmt-cluster.xml]: Fixed.

File: src/app/zap-templates/zcl/data-model/chip/group-key-mgmt-cluster.xml (hunk 1 above)
Comment (rcasallas-silabs) [reply src/app/zap-templates/zcl/data-model/chip/group-key-mgmt-cluster.xml]: Fixed.

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**